src/extension/feature-pregeneralized/src/main/java/org/geoserver/data/gen/info/GeneralizationInfosProviderImpl.java

```diff
@@ -50,4 +58,41 @@ protected URL deriveURLFromSourceObject(Object source) throws IOException {
         }
         throw new IOException("Cannot read from " + source);
     }
+
+    @Override
+    protected GeneralizationInfos parseXML(URL url) throws IOException {
+
+        File configurationFile =
+                ((GeoServerResourceLoader) GeoServerExtensions.bean("resourceLoader"))
+                        .fromURL(url)
+                        .file();
+
+        Document doc = null;
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        factory.setIgnoringComments(true);
+        factory.setNamespaceAware(true);
+        factory.setIgnoringElementContentWhitespace(true);
+
+        try {
+            DocumentBuilder db = factory.newDocumentBuilder();
+            doc = db.parse(configurationFile);
```

Comment:
Does DocumentBuilder have option to take an inputStream? resource.in() could then be used

Reply:
Yes the parse method exists with the inputStream: I tried to use it but I got each time a "premature end of file" error. I will try it again since I have already a first working implementation.

Reply:
Changed in the last commit.

Reply:
that is great!